packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -287,7 +287,11 @@ export function getValidatorApi({
     // as of now fee recipient checks can not be performed because builder does not return bid recipient
     {
       skipHeadChecksAndUpdate,
-    }: Omit<routes.validator.ExtraProduceBlockOps, "builderSelection"> & {skipHeadChecksAndUpdate?: boolean} = {}
+      commonBlockBody,
+    }: Omit<routes.validator.ExtraProduceBlockOps, "builderSelection"> & {
+      skipHeadChecksAndUpdate?: boolean;
+      commonBlockBody?: CommonBlockBody;
```

Comment:
why do we type this as optional while we always provide value and we have to handle fallback logic downstream?

Reply:
We don't always provide option, one can directly call `chain.produceBlock` and it will still be able to produce the block. We are doing it in our code for `produceBlockV2`